### DISCUSSION TITLE
Identify corrupted entries before reading their data

### DISCUSSION THREAD

**Discussion by ksidlauskas** [gildas-lormeau/zip.js#261] [Q&A] [answered]
Hi, is it possible to identify which files are corrupted inside the zip before you start reading their data?

I have a zip file that is causing issues when extracting (local file header not found), but I would like to know that at the time i check the contents of the zip, rather than when i'm already trying to read the data.

I compared the entry information for all the contents of the zipfile, but they don't seem to have any obvious issues.

The error is thrown on `const blob = await file.getData(new zip.BlobWriter(file.type));`
below is the error info that's logged out. I am using the zip-full.js;

![image](https://user-images.githubusercontent.com/5751651/124165291-4319ec80-daaa-11eb-860e-7f7c5e04fc1f.png)

**Comment by gildas-lormeau** [comment, marked as answer]
The local header of an entry in a zip file must begin with `0x504b0304` (in big-endian). The error you mentioned is thrown when this value is not found (cf. chapter 4.3.7 in the [spec](https://pkware.cachefly.net/webdocs/APPNOTE/APPNOTE-6.3.9.TXT#:~:text=4.3.7%20%20Local%20file%20header%3A)).
https://github.com/gildas-lormeau/zip.js/blob/a8f01892cb0fe93c99baf439aaf0eba2f8508cf7/lib/core/zip-reader.js#L260-L262

This test cannot be run when listing entries because the `ZipReader#getEntries` method would run much slower, especially when retrieving the zip content with a `HttpRangeReader` instance or when reading a zip file with hundreds of entries.

Note however that you can abort the execution of `Entry#getData` if you want to test an entry without reading the data. For this, you can use the `signal` option, e.g.

```js
const entries = await zipReader.getEntries();
const controller = new AbortController();
try {
  const promiseEntryData = entries[0].getData(new zip.BlobWriter(), { signal: controller.signal });
  controller.abort();
  await promiseEntryData;
} catch (error) {
  if (error.message != zip.ERR_ABORT) {
    throw error;
  }
}
```
